packages/core/src/common/keybinding.ts

```diff
@@ -311,37 +338,66 @@ export class KeybindingRegistry {
         }
 
         const keyCode = KeyCode.createKeyCode(event);
-        const bindings = this.getKeybindingsForKeyCode(keyCode);
+        /* Keycode is only a modifier, next keycode will be modifier + key.
+           Ignore this one.  */
+        if (keyCode.isModifierOnly()) {
+            return;
+        }
 
-        for (const binding of bindings) {
-            const context = binding.context
-                ? this.contextRegistry.getContext(binding.context)
-                : undefined;
+        if (this.timeoutTimer) {
+            clearTimeout(this.timeoutTimer);
+        }
 
-            /* Only execute if it has no context (global context) or if we're in
-               that context.  */
-            if (!context || context.isEnabled(binding)) {
+        this.keySequence.push(keyCode);
+        const bindings = this.getKeybindingsForKeySequence(this.keySequence);
 
-                if (this.isPseudoCommand(binding.command)) {
-                    /* Don't do anything, let the event propagate.  */
-                } else {
-                    const command = this.commandRegistry.getCommand(binding.command);
-                    if (command) {
-                        const commandHandler = this.commandRegistry.getActiveHandler(command.id);
+        if (bindings.full.length > 0) {
```

Comment:
I think it will be better to extract a function that returns true if a command was executed and false if not, in the second case we will try to match partial bindings

Reply:
OK it does make for a long function here.

Reply:
I was more concern about mixing code of execution command and handling partial bindings, such function won't need to care about them.
```js
if (fn(bindings)) {
    this.keySequence = [];
} else if (bindings.partial.length > 0) {
    ...
}
```

Reply:
yes it helps with that too, I'll refactor that a bit